realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -144,6 +145,39 @@ private SyncConfiguration(File directory,
         this.waitForInitialData = waitForInitialData;
     }
 
+    /**
+     * Returns a SyncConfiguration appropriate to open a synced Realm offline.
+     * This is useful when trying to open a backup Realm (after a Client Reset)
+     *
+     * @param canonicalPath the absolute path to the Realm file defined by this configuration.
+     * @param encryptionKey the key used to encrypt/decrypt the Realm file.
+     * @param modules if specified it will restricts Realm schema to the provided module.
+     * @return SyncConfiguration that can be used offline
+     */
+    public static RealmConfiguration forOffline(String canonicalPath, @Nullable byte[] encryptionKey, Object... modules) {
```

Comment:
Uggh, the signature for this method :( ... I can see why it is needed, but still.

Any reason to specify modules? At worst, we will open the Realm with a superset of the classes available? This will create those in the Realm file, but is that a big problem?

Should `modules` be nullable? Would also require a null check in the code

I know this is really a power feature, but should we also have a simple variant that is just `forOffline(String path)` ?